rts/Lua/LuaTracyExtra.h

```diff
@@ -0,0 +1,26 @@
+/* This file is part of the Recoil engine (GPL v2 or later), see LICENSE.html */
+
+#pragma once
+
+#include <functional>
+#include <set>
+#include <string>
+
+struct lua_State;
+
+class LuaTracyExtra {
```

Comment:
Why it's a class with all members just being static? Why not just namespace if you want to group things together under a name?

Also why `tracyLuaPlots` does need to be defined in include file? It's not part of the interface, it's static global, and now two different translation units need to parse more header files.

Reply:
> why class

I ~~brainlessly copypasted~~ _followed the precedent_ of `LuaMathExtra`. Want me to make them both into a namespace?

> why `tracyLuaPlots` member

No particular reason, I usually just put globals as static members of relevant classes because that makes it a bit more convenient to refactor later if multiple instances of the var are needed (e.g. for unit tests; I guess this also sort of answers why class and not just a namespace). But yes, in this case maybe it doesn't actually help and just makes compilation more annoying. I can move to the implementation file.

Reply:
I'm not sure how strong is this precedence in the engine code base but in this isolated case it IMHO it would be cleaner to either make it a singleton or use regular functions maybe in namespace. Even `inline static` is a C++17 feature I had to look up documentation to make sure I understand how it works.